app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/SearchWorkerResult.kt

```diff
@@ -92,65 +55,51 @@ import com.arygm.quickfix.model.quickfix.QuickFixViewModel
-import java.time.LocalDate
 
-data class SearchFilterButtons(
-    val onClick: () -> Unit,
-    val text: String,
-    val leadingIcon: ImageVector? = null,
-    val trailingIcon: ImageVector? = null,
-    val applied: Boolean = false
-)
-
-@SuppressLint("SuspiciousIndentation")
 @OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
 @Composable
 fun SearchWorkerResult(
     navigationActions: NavigationActions,
     searchViewModel: SearchViewModel,
     accountViewModel: AccountViewModel,
     userProfileViewModel: ProfileViewModel,
-    preferencesViewModel: PreferencesViewModel,
     quickFixViewModel: QuickFixViewModel,
-    geocoderWrapper: GeocoderWrapper = GeocoderWrapper(LocalContext.current),
+    preferencesViewModel: PreferencesViewModel,
     workerViewModel: ProfileViewModel,
-    locationHelper: LocationHelper = LocationHelper(LocalContext.current, MainActivity())
 ) {
-  fun getCityNameFromCoordinates(latitude: Double, longitude: Double): String? {
-    val addresses = geocoderWrapper.getFromLocation(latitude, longitude, 1)
-    return addresses?.firstOrNull()?.locality
-        ?: addresses?.firstOrNull()?.subAdminArea
-        ?: addresses?.firstOrNull()?.adminArea
-  }
-  var phoneLocation by remember {
-    mutableStateOf(com.arygm.quickfix.model.locations.Location(0.0, 0.0, "Default"))
-  }
-  var baseLocation by remember { mutableStateOf(phoneLocation) }
+  val (uiState, setUiState) = remember { mutableStateOf(SearchUIState()) }
+  var isWindowVisible by remember { mutableStateOf(false) }
   val context = LocalContext.current
+  val locationHelper = LocationHelper(context, MainActivity())
+  var selectedWorkerProfile by remember { mutableStateOf(WorkerProfile()) }
+  val filterState = rememberSearchFiltersState()
+  val defaultBitmap =
+      Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888) // Example default Bitmap
+
+  var profilePicture by remember { mutableStateOf(defaultBitmap) }
+  var bannerPicture by remember { mutableStateOf(defaultBitmap) }
+  var baseLocation by remember { mutableStateOf(filterState.phoneLocation) }
   var userProfile by remember { mutableStateOf<UserProfile?>(null) }
   var uid by remember { mutableStateOf("Loading...") }
+  val searchSubcategory by searchViewModel.searchSubcategory.collectAsState()
+
+  // Fetch user and set base location
 
   var loading by remember { mutableStateOf(true) } // Tracks if data is loading
 
   LaunchedEffect(Unit) {
     if (locationHelper.checkPermissions()) {
       locationHelper.getCurrentLocation { location ->
         if (location != null) {
-          phoneLocation =
-              com.arygm.quickfix.model.locations.Location(
-                  location.latitude, location.longitude, "Phone Location")
-          baseLocation = phoneLocation
+          val userLoc = Location(location.latitude, location.longitude, "Phone Location")
+          filterState.phoneLocation = userLoc
         } else {
```

Comment:
Why the change ?

Reply:
because I found it more intuitive to have the phone location in the filterState but I can change it if you feel that it is a bad idea :)